Busybox no longer builds

**Describe the bug**
busy box does not build anymore on master branch, It seems to relate on #67644 

**To Reproduce**
Install busybox

**Expected behavior**
Successfully installing

**Screenshots**
```
/nix/store/cl1i6bfqnx48ipakj4px7pb1babzs23j-binutils-2.31.1/bin/ld: cannot find -lcrypt
/nix/store/cl1i6bfqnx48ipakj4px7pb1babzs23j-binutils-2.31.1/bin/ld: cannot find -lm
collect2: error: ld returned 1 exit status
Note: if build needs additional libraries, put them in CONFIG_EXTRA_LDLIBS.
Example: CONFIG_EXTRA_LDLIBS="pthread dl tirpc audit pam"
make: *** [Makefile:718: busybox_unstripped] Error 1
builder for '/nix/store/m2x5ln38d2qjdr59y33mvzgnbzavpan5-busybox-1.30.1.drv' failed with exit code 2
cannot build derivation '/nix/store/a9rfb5nl6d7i663phh4i351n8il0x4m0-nix-2.3.drv': 1 dependencies couldn't be built
error: build of '/nix/store/a9rfb5nl6d7i663phh4i351n8il0x4m0-nix-2.3.drv' failed
```

**Additional context**
Add any other context about the problem here.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.1, NixOS, 20.03.git.e300b1c (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(shyim): `"home-manager"`
 - channels(root): `"nixos-19.03.173552.bd5e8f35c2e, unstable-20.03pre194293.2436c27541b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: busybox
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/os-specific/linux/busybox/default.nix
+++ b/pkgs/os-specific/linux/busybox/default.nix
@@ -1,7 +1,7 @@
 { stdenv, lib, buildPackages, fetchurl
 , enableStatic ? false
 , enableMinimal ? false
-, useMusl ? stdenv.hostPlatform.libc == "musl"
+, useMusl ? stdenv.hostPlatform.libc == "musl", musl
 , extraConfig ? ""
 }:
 
@@ -86,6 +86,10 @@ stdenv.mkDerivation rec {
     make oldconfig
 
     runHook postConfigure
+  '';
+
+  postConfigure = lib.optionalString useMusl ''
+    makeFlagsArray+=("CC=${stdenv.cc.targetPrefix}cc -isystem ${musl.dev}/include -B${musl}/lib -L${musl}/lib")
   '';
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];

